crates/full-node/sov-sequencer/src/preferred/cache_warm_up_executor.rs

```diff
@@ -114,21 +150,25 @@ impl<S: Spec> CacheWarmUpExecutor<S> {
                     }
 
                     tx = tx_receiver.recv_async() => {
-                        let baked_tx = match tx {
+                        let tx_with_sender = match tx {
                              Ok(tx) => tx,
                              Err(flume::RecvError::Disconnected) => {
                                 // Quit if channel closed.
                                 return
                             },
                         };
                         if is_started {
-                            let res = executor.apply_tx_to_in_progress_batch(&baked_tx).await;
+                            let baked_tx = FullyBakedTxWithMaybeChangeSet::new(tx_with_sender.tx);
+                            let res = executor.apply_tx_to_in_progress_batch(baked_tx).await;
 
                             match res{
-                                Ok((_, _tx_change_set)) => {
-                                    // TODO
+                                Ok((_, tx_change_set)) => {
+                                    // It ok safe to ignore the error if the receiver was dropped.
+                                    // This can happen if the transaction on the main executor has already finished.
+                                    let _ = tx_with_sender.sender.send(tx_change_set);
```

Comment:
Here it looks like we're sending the changes. For full optimistic execution we'll need the writes as well - but for the current PR we want to send the *reads* right?

Reply:
In the first version I want to handle only reads according to our previous message:
```
For each read in the set from the channel…
Check if the key has already been read/written. If so, ignore it
Put the read value into the cache
```
Once we start working on the full version, we will add the writes, but I would like to close the whole logical flow.